### PR TITLE
Move tox to pytest

### LINKED ISSUE #137: Add pytest to github actions
### Expected Behavior

Add pytest to github actions

### Current Behavior

Only style testing is done as of the moment.

### PR DESCRIPTION
**Pull request by alejandrosame**
## Description
Fixes  #137 and #192. Apart from substituting tox with pytest testing, Makefile and Github workflow files have been aligned for consitency. Since right now, we lack meaningful Python code, coverage dependency and code has been deleted also for simplicity.

## Affected Dependencies
Tox deleted. Coverage deleted.

## How has this been tested?
- Locally, I ran `make test`.
- Github Actions were tested on a mock PR: https://github.com/alejandrosame/PyDP/pull/3

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by chinmayshah99** [comment]
@alejandrosame the tests are failing. 
Also, is this tested on a system with conda installed?

**Comment by alejandrosame** [comment]
@chinmayshah99 If you check the log, the error is related to the current bug we have with bounds. If you want, I can already deactivate the tests for bounded methods in this PR since it is more or less a related issue (having consistent tests ASAP). 

Regarding conda, they have not been tested using conda. As the changes show, pipenv was added to Github actions for consistency with Docker and host execution. I think validating conda support should be a separate issue/PR (it will require also more documentation updates, new github actions, etc).

**Comment by chinmayshah99** [comment]
Validating with conda is a must, since a lot of our folks use conda.

Regards

Chinmay Shah

File: setup.cfg [hunk 1]
```diff
@@ -13,7 +13,7 @@ replace = __version__ = "{new_version}"
 
 [pycodestyle]
```
Comment (chinmayshah99) [inline setup.cfg]: Is this needed now that we have removed `pycodestyle`?

**Comment by chinmayshah99** [comment]
Error when running `make test`
```
cs@cs-VirtualBox:~/Desktop/PyDP$ make test
pipenv run black ./ --check --diff
All done! ✨ 🍰 ✨
22 files would be left unchanged.
pipenv run ./run-clang-format.py -r src/bindings/
Traceback (most recent call last):
  File "/home/cs/.local/bin/pipenv", line 8, in <module>
    sys.exit(cli())
  File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 696, in main
    _verify_python3_env()
  File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/_unicodefun.py", line 124, in _verify_python3_env
    ' mitigation steps.' + extra
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment. Consult https://click.palletsprojects.com/en/7.x/python3/ for mitigation steps.

This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8
Makefile:56: recipe for target 'check-style-cpp' failed
make: *** [check-style-cpp] Error 1
```
Though I can add these environment variables, this error is not desirable.

File: setup.cfg (hunk 1 above)
Comment (alejandrosame) [reply setup.cfg]: Indeed it is not needed since we introduced Black. I'll delete the section.

**Comment by alejandrosame** [comment]
> Error when running `make test`
> 
> ```
> cs@cs-VirtualBox:~/Desktop/PyDP$ make test
> pipenv run black ./ --check --diff
> All done! ✨ 🍰 ✨
> 22 files would be left unchanged.
> pipenv run ./run-clang-format.py -r src/bindings/
> Traceback (most recent call last):
>   File "/home/cs/.local/bin/pipenv", line 8, in <module>
>     sys.exit(cli())
>   File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 764, in __call__
>     return self.main(*args, **kwargs)
>   File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 696, in main
>     _verify_python3_env()
>   File "/home/cs/.local/lib/python3.6/site-packages/pipenv/vendor/click/_unicodefun.py", line 124, in _verify_python3_env
>     ' mitigation steps.' + extra
> RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment. Consult https://click.palletsprojects.com/en/7.x/python3/ for mitigation steps.
> 
> This system supports the C.UTF-8 locale which is recommended.
> You might be able to resolve your issue by exporting the
> following environment variables:
> 
>     export LC_ALL=C.UTF-8
>     export LANG=C.UTF-8
> Makefile:56: recipe for target 'check-style-cpp' failed
> make: *** [check-style-cpp] Error 1
> ```
> 
> Though I can add these environment variables, this error is not desirable.

I added an ENV file so Pipenv automatically loads those variables inside the virtualenv. That should fix the issue. Can you please confirm that it fixes the execution of `make test` in your system?

**Comment by chinmayshah99** [comment]
> I added an ENV file so Pipenv automatically loads those variables inside the virtualenv. That should fix the issue. Can you please confirm that it fixes the execution of `make test` in your system?

This works well! Just to confirm, there is no-inplace editing right? The code does not automatically re-format the correct code even though it suggests what needs to be done. Can we add that if it's possible?

**Comment by alejandrosame** [comment]
> > I added an ENV file so Pipenv automatically loads those variables inside the virtualenv. That should fix the issue. Can you please confirm that it fixes the execution of `make test` in your system?
> 
> This works well! Just to confirm, there is no-inplace editing right? The code does not automatically re-format the correct code even though it suggests what needs to be done. Can we add that if it's possible?

I wouldn't overload `make test` with in-place editing. In my opinion, the easiest workflow is to hook the relevant formatting functions to the editor, so the code is formatted every time is saved. That's helpful because one gets to see the actual expected formatting in the editor, and, in my opinion, that helps getting used to the style. Therefore, I think we should notify this to contributors, so they know that possibility exists.

Apart from that, for those that can't arrange the editor (or wouldn't care because, for example, they just want to quickly make a one time contribution) we could provide new formatting actions. That way, `make test` can announce that they should run, for example, `make format-code` to automatically format the faulty files. 

What do you think?

**Comment by chinmayshah99** [comment]
> Apart from that, for those that can't arrange the editor (or wouldn't care because, for example, they just want to quickly make a one time contribution) we could provide new formatting actions. That way, `make test` can announce that they should run, for example, `make format-code` to automatically format the faulty files.
> 
> What do you think?

Yeah, we can add that! 

Also can we also add `make build` that just builds/ run the `./build_pydp.sh`?

**Comment by alejandrosame** [comment]
> Yeah, we can add that!
> 
> Also can we also add `make build` that just builds/ run the `./build_pydp.sh`?

Done on the last 2 commits.

**Review by chinmayshah99 — APPROVED** [review]
LGTM! Need to confirm if all tests run perfectly :)

**Pull request merged**